File: src/frontend/models/document.dart
class Document {
  final int id;
  final String title;
  final String content;
  final int userId;

  Document({
    required this.id,
    required this.title,
    required this.content,
    required this.userId
  });

  factory Document.fromJson(Map<String, dynamic> json) {
    return Document(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      userId: json['user_id']
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'title': title,
    'content': content,
    'user_id': userId
  };
}

File: src/backend/core/models/document.py
from backend.core.models.base import BaseModel
from sqlalchemy import Column, String, Text

class Document(BaseModel):
    __tablename__ = "documents"

    user_id = Column(String, index=True)  # Reference to auth service
    title = Column(String, index=True)
    content = Column(Text)
